sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NestedCodeBlock.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void Method(int value)
+        { // Compliant, parent is MethodDeclaration
```

Comment:
I would also add **compliant** tests with nested

- `using`,
- `try-catch-finally`
- `do-while`
- `foreach`
- `unsafe { }`
- `checked {}`
- `lock (this) {}`
- `unchecked { }`
- `event EventHandler<LoadEventArgs> LoadedData = delegate { };` (on this one I'm note sure it 
makes sense, so if it doesn't skip it)
- _enum_
- a local function
```
public void Method1()
{
  var x = LocalFoo();
  bool LocalFoo()
  {
    return true;
  }
}
```

Also, **Noncompliant** tests with nested blocks inside
- _constructor_
- _property (setter and getter)_
- _struct_